tests/backwards_compatibility_tests/compatibility_test_runner.py

```diff
@@ -615,6 +525,188 @@ def copy_state_from_container(
+        logger.debug(f"Starting marqo to_version: {to_version} container by transferring state from version {from_version} to {to_version}")
+        start_marqo_container_by_transferring_state(to_version, from_version, from_version_volume,
+                                                    to_version_image_idenfitifer, "ECR")
+        logger.debug(f"Started marqo to_version: {to_version} container by transferring state")
+        # Step 4: Run tests
+        try:
+            run_tests_across_versions("test", from_version)
+        except Exception as e:
+            logger.error(f"Error running tests across versions in 'test' mode: {e}")
+            raise e
+        logger.debug("Finished running tests in Test mode")
+        # Step 5: Do a full test run which includes running tests in prepare and test mode on the same container
+        try:
+            full_test_run(to_version)
+        except Exception as e:
+            logger.error(f"Error running tests in full test run: {e}")
+            raise e
+    except Exception as e:
+        logger.debug(f"An error occurred while executing backwards compatibility tests: {e}")
+        raise e
+    finally:
+        # Stop the to_version container (but don't remove it yet)
+        logger.debug("Calling stop_marqo_container with " + str(to_version))
+        stop_marqo_container(to_version)
+        # Clean up all containers at the end
+        cleanup_containers()
+        cleanup_volumes()
+
+
+
+def rollback_test(to_version: str, from_version: str, to_version_image_idenfitifer: str):
+    """
+    Perform a rollback test between two versions of Marqo.
+    #TODO: fix the comment
+    This function first runs test cases in prepare mode on from_version marqo container, then upgrades it to to_version marqo container,
+    It then downgrades (rollback) to from_version container again where it runs test cases in test mode. Finally, it triggers rollback endpoint
+    to rollback vespa application and runs the complete test suite again.
+
+    Args:
+        to_version (str): The target version of the Marqo container.
+        from_version (str): The source version of the Marqo container.
+        to_version_image_idenfitifer (str): The unique identifier for a to_version image. It can be either be the fully qualified image name with the tag
+    """
+    logger.debug(f"Starting marqo rollback tests with from_version: {from_version}, to_version: {to_version}, to_version_image_idenfitifer: {to_version_image_idenfitifer}")
+    try:
+        # Step 0: Generate a volume name to be used with the "from_version" Marqo container for state transfer.
+        from_version_volume = get_volume_name_from_marqo_version(from_version)
+
+        # Step 1: Start a Marqo container using from_version and run tests in prepare mode.
+        start_marqo_container(from_version, from_version_volume)
+        logger.debug(f"Started marqo container {from_version}")
```

Comment:
nit I'm on the fence with these log levels. They feel like info messages. I also see you had to change your default log level to debug